refactor(master): Refactor node operations

lgsilva3087 opened the pull request:
This PR improves the code related to operations on top of nodes.
The commits introduce the following changes:

- rename FS node operations.
- rename FS node operations arguments.
- rename private node operations.
- use std::array in checkFile function.
- use std::array for eattr counters.

Please refer to the individual commits for extended descriptions.

Reviewers: I am pausing the refactor at this point to make the review easier/shorter, but I will continue immediately with it.

lgsilva3087 committed:
refactor(master): rename FS node operations

Unifies function names in IFilesystemNodeOperations to follow the
project's camelCase convention and shorten identifiers.

Updates FilesystemNodeOperationsBase and all call sites accordingly.

Signed-off-by: guillex <guillex@leil.io>

lgsilva3087 committed:
refactor(master): Rename FS node operations arguments

This commit improves the argument names for several functions of
IFilesystemNodeOperations interface and the implementation
FilesystemNodeOperationsBase. The main changes are:

- More descriptive names.
- Change some of them to camelCase to match our style.
- Remove clang-tidy warnings about names too short.

Signed-off-by: guillex <guillex@leil.io>

lgsilva3087 committed:
refactor(master): rename private node operations

Improves naming of private functions of FilesystemNodeOperationsBase
and their arguments, for better readability, self-documentation and
better match to our code style.

Signed-off-by: guillex <guillex@leil.io>

lgsilva3087 committed:
refactor(master): Use std::array in checkFile function

Replace C-style array (`uint32_t chunkcount[CHUNK_MATRIX_SIZE]`) with
modern and safer std::array, introducing the `ChunkCountArray` type
alias.

Signed-off-by: guillex <guillex@leil.io>

lgsilva3087 committed:
refactor(master): use std::array for eattr counters

Replace raw C-style arrays (uint32_t[16]) with a type-safe
std::array-based ExtendedAttributesArray in extended attributes
handling functions. This improves type safety, eliminates magic
numbers, and provides better semantics through named type aliases.

Also suppress clang-tidy warnings.

Signed-off-by: guillex <guillex@leil.io>

dmga44 commented on src/master/filesystem_operations_interface.h:
```diff
@@ -165,13 +165,14 @@ class IFilesystemOperations {
 	virtual uint8_t getEAttr(const FsContext &context, inode_t inode, uint8_t gmode,
-	                         uint32_t feattrtab[16], uint32_t deattrtab[16]) = 0;
+	                         ExtendedAttributesArray &fileEAttrTab,
+	                         ExtendedAttributesArray &dirEAttrTab) = 0;
```
Please note these are extra attributes, see saunafs-geteattr.1.adoc. Consider renaming the parameters.

dmga44 commented on src/master/matoclserv.cc:
```diff
@@ -3055,9 +3055,8 @@ void matoclserv_fuse_setgoal(matoclserventry *eptr, PacketHeader header, const u
-	uint32_t feattrtab[kMaxEattr];
-	uint32_t deattrtab[kMaxEattr];
+	ExtendedAttributesArray fileEAttrTab;
+	ExtendedAttributesArray dirEAttrTab;
```
Consider renaming as mentioned above.

dmga44 commented on src/master/matoclserv.cc:
```diff
@@ -3102,16 +3101,16 @@ void matoclserv_fuse_geteattr(matoclserventry *eptr, const uint8_t *data, uint32
+			if (dirEAttrTab[i]) {
 				put8bit(&ptr, i);
-				put32bit(&ptr, deattrtab[i]);
+				put32bit(&ptr, dirEAttrTab[i]);
```
Consider, as mentioned before, that these are extra attributes.

rolysr commented on src/master/filesystem_operations.cc:
```diff
@@ -612,19 +609,19 @@ uint8_t FilesystemOperationsBase::getAttr(const FsContext &context, inode_t inod
 
-	nodeOperations_->fsnodes_fill_attr(p, NULL, context.uid(), context.gid(), context.auid(),
-	                                   context.agid(), context.sesflags(), attr);
+	nodeOperations_->fillAttr(p, NULL, context.uid(), context.gid(), context.auid(), context.agid(),
```
Consider changing here the `NULL` by `nullptr` as in other modified parts if possible.

lgsilva3087 replied:
I need to refactor properly the related .cc files.
In that file there are at least 11 ocurrences of NULL.

rolysr commented on src/master/filesystem_node.cc:
```diff
@@ -895,136 +885,126 @@ uint32_t FilesystemNodeOperationsBase::getDetachedSize(const ReservedPathContain
+					::memcpy(outBuffer, attr.data(), attr.size());
 				} else {
-					FSNode *rn = idToNode(rootinode);
+					FSNode *rn = idToNode(rootINode);
```
Consider renaming this rn here also.

lgsilva3087 replied:
Yes, I will do it in the next steps.

rolysr commented on src/master/filesystem_node.cc:
```diff
@@ -1342,56 +1318,55 @@ void FilesystemNodeOperationsBase::fsnodes_remove_node(uint32_t ts, FSNode *node
-			child->ctime = ts;
-			fsnodes_update_checksum(child);
+	if (childNode->type == FSNodeType::kFile) {
+		auto *file_node = static_cast<FSNodeFile *>(childNode);
```
Consider camel casing this `file_node` variable.

lgsilva3087 replied:
Yes, I will do it in the next steps.

rolysr commented on src/master/filesystem_node.cc:
```diff
@@ -115,61 +111,55 @@ bool FilesystemNodeOperationsBase::last_chunk_nonempty(FSNodeFile *node) {
-			uint64_t size = (uint64_t)nonzero_chunks * full_chunk_realsize;
-			if (last_chunk_nonempty(node)) {
+			    ecChunkRealSize(SFSBLOCKSINCHUNK, data_part_count, parity_part_count);
+			uint64_t size = (uint64_t)nonZeroChunks * full_chunk_realsize;
```
Consider using modern cast here.

lgsilva3087 replied:
Yes, I will do it in the next steps.

rolysr reviewed (approved):
Thanks for this work @lgsilva3087. Consider some of my comments as optional as in many cases they could be not directly related with the intended change on the corresponding commit.

lgsilva3087 committed:
refactor(master): rename eattr operations to extraattr

Rename filesystem node operations and type aliases to use consistent
"ExtraAttr" terminology instead of "EAttr" to better distinguish from
POSIX extended attributes (xattr).

Co-authored-by: Dave <dave@leil.io>

Signed-off-by: guillex <guillex@leil.io>

dmga44 reviewed (approved):
Great job @lgsilva3087 !

Please check my minor comments.
